llmfoundry/models/hf/hf_prefix_lm.py

```diff
@@ -145,7 +144,7 @@ def __init__(self, om_model_config: DictConfig, tokenizer: Tokenizer):
 
         return composer_model
 
-    def forward(self, batch):
+    def forward(self, batch: MutableMapping):
```

Comment:
I've never seen this type applied to a `batch` before, might be confusing to users. Is there a more narrow type possible here?

Reply:
So, I _think_ this is the right type although open to a different suggestion. Sometimes the HF batch input was a `UserDict` (which is why this isn't just a `Dict` type). A `Union[Dict, UserDict]` probably also works, but `Mapping` does encompass the properties needed for a batch. As for the `MutableMapping` vs `Mapping`, they could all be `MutableMapping` without issue. Just set them based on whether the code in question tried to modify the batch or not.